src/main/java/org/ow2/proactive/connector/iaas/cloud/provider/azure/AzureProvider.java

```diff
@@ -0,0 +1,602 @@
+
+        // Set the credentials (whether password or SSH key)
+        return optionalPublicKey.map(creatableVMWithoutCredentials::withSsh)
+                                .orElseGet(() -> creatableVMWithoutCredentials.withRootPassword(optionalPassword.orElse(defaultPassword)));
+    }
+
+    private VirtualMachine.DefinitionStages.WithWindowsCreateManaged configureWindowsVirtualMachine(Azure azureService,
+            String instanceTag, Region region, ResourceGroup resourceGroup, InstanceCredentials instanceCredentials,
+            VirtualMachineCustomImage image, Creatable<NetworkInterface> creatableNetworkInterface) {
+        // Retrieve optional credentials
+        Optional<InstanceCredentials> optionalInstanceCredentials = Optional.ofNullable(instanceCredentials);
+        Optional<String> optionalUsername = optionalInstanceCredentials.isPresent() ? Optional.ofNullable(optionalInstanceCredentials.get()
+                                                                                                                                     .getUsername())
+                                                                                    : Optional.empty();
+        Optional<String> optionalPassword = optionalInstanceCredentials.isPresent() ? Optional.ofNullable(optionalInstanceCredentials.get()
+                                                                                                                                     .getPassword())
+                                                                                    : Optional.empty();
+
+        // Prepare the VM with credentials
+        return azureService.virtualMachines()
+                           .define(instanceTag)
+                           .withRegion(region)
+                           .withExistingResourceGroup(resourceGroup)
+                           .withNewPrimaryNetworkInterface(creatableNetworkInterface)
+                           .withWindowsCustomImage(image.id())
+                           .withAdminUsername(optionalUsername.orElse(defaultUsername))
+                           .withAdminPassword(optionalPassword.orElse(defaultPassword));
+    }
+
+    /**
+     * Create a uniq tag for a VM based on the original tag provided and the instance index
+     *
+     * @param tagBase       the tag base
+     * @param instanceIndex the instance index
+     * @return a uniq VM tag
+     */
+    private static String createUniqInstanceTag(String tagBase, int instanceIndex) {
+        if (instanceIndex > 1) {
+            return tagBase + String.valueOf(instanceIndex);
+        }
+        return tagBase;
+    }
+
+    private static String createUniqSecurityGroupName(String instanceTag) {
+        return SdkContext.randomResourceName(instanceTag + '-' + networkSecurityGroupNameBase,
```

Comment:
Well good question, I assume that the 'RESOURCES_NAME_EXTRA_CHARS' need to be set high enough to minimize collisions risks. We could check all resources' name to be sure but this may considerably slowdow the deployment.
I changed 'RESOURCES_NAME_EXTRA_CHARS' from 5 to 10 to minimize collision risks in 8d57d05